src/pages/sitemap.xml.tsx

```diff
@@ -0,0 +1,53 @@
+import { GetServerSideProps } from 'next'
+import { collectedNotes } from "collected-notes";
+
+import React from 'react'
+
+const Sitemap: React.FC = () => null
+
+const cn = collectedNotes(process.env.CN_EMAIL, process.env.CN_TOKEN);
+
+export const getServerSideProps: GetServerSideProps = async ({ res }) => {

```

Comment:
This should be an API route, you can setup in `vercel.json` a rewrite to point `/sitemap.xml` to the `/api/sitemap` route